package/MDAnalysis/analysis/Aggregation_size.py

```diff
@@ -0,0 +1,364 @@
+    3- Find the aggregates by checking only one time each using the
+    "aggregate" subfunction (to be efficient)
+    4- Check and merge two aggregates as one aggregate if the distance
+    between any element of them is less than cut_off
+    5- Eliminate monomers (optional)
+    6- measure average gyration radius (optional)
+    7- measure the number of aggregates with different size
+    """
+    import warnings
+    import numpy as np
+    import MDAnalysis as mda
+    from MDAnalysis.lib import distances
+    u = mda.Universe(gro_file, xtc_file)
+    molnum = number_of_molecules
+    Limit = cut_off
+    i = 0
+    for it in reversed(range(1, molnum)):
+        i += 1
+        globals()["d" + str(i)] = np.zeros(
+            it * len(u.trajectory)).reshape(it, len(u.trajectory))
+    if Type == "Closest":
+        cut_off = cut_off or 3.5
+        j_time = 0
+        for ts in u.trajectory:
+            for i in range(1, molnum+1):
+                ii = 0
+                for j in range(i+1, molnum+1):
+                    residue1 = u.select_atoms(f"resid {i}").positions
+                    residue2 = u.select_atoms(f"resid {j}").positions
+                    dists = distances.distance_array(
+                        residue1, residue2, box=u.universe.dimensions,
+                        backend='OpenMP')
+                    globals()["d" + str(i)][ii][j_time] = np.amin(dists)
+                    ii += 1
+            j_time += 1
+    elif Type == "COM":
+        cut_off = cut_off or 8.5
+        j_time = 0
+        for ts in u.trajectory:
+            cm = {}
+            for x in range((molnum)):
+                cm.update(
+                    {x+1:u.select_atoms(f"resid {x+1}").center_of_mass()})
+            cm_coordintes = np.array(list(cm.values()))
+            dists = distances.distance_array(
+                cm_coordintes, cm_coordintes, box=u.universe.dimensions)
+            for j in range(molnum):
+                ii = 0
+                for i in range(j+1, molnum):
+                    globals()["d" + str(j+1)][ii][j_time] = dists[i][j]
+                    ii += 1
+            j_time += 1
+    elif Type == "Specific":
+        if (not(atom_1 and atom_2)):
+            warnings.warn("enter valid atom_1 and atom_2")
+        cut_off = cut_off or 8.5
+        j_time = 0
+        for ts in u.trajectory:
+            residue1 = u.select_atoms(f"name {atom_1}").positions
+            residue2 = u.select_atoms(f"name {atom_2}").positions
+            dists = distances.distance_array(
+                residue1, residue2, box=u.universe.dimensions)
+            for j in range(molnum):
+                ii = 0
+                for i in range(j+1, molnum):
+                    globals()["d" + str(j+1)][ii][j_time] = dists[i][j]
+                    ii += 1
+            j_time += 1
+    else:
+        warnings.warn("enter valid Type")
+    GYR = []
+    ij = 0
+    for ts in u.trajectory:
+        u.trajectory[ij]
+        ij += 1
```

Comment:
Here, you are doing a lot of redundant gymnastic. In MDAnalysis, when you iterate over a trajectory, you move the cursor so that the trajectory points at the frame you requested. When you index the trajectory with `u.trajectory[ij]`, you do the same thing. So here, you move to frame 0, `ts` corresponds to frame 0 so `ts.frame` is 0; then you move again to frame 0 by doing `u.trajectory[ij]`; because we sometimes but not always reuse the time step object, `ts` may or may not be the same as previously, but because you moved to the same frame `ts.frame` is 0 anyway, same as `ij`, same as `tss`.

One way of avoiding the redundant operations would be to only do:

```python
for ts in u.trajectory:
    tss = ts.frame
   ...
```

Her you save yourself the hassle and the time it takes to jump back to the frame you are already at.

Tough, I do not think you are using the fact that the trajectory actually advances in this loop, and you only want to run your loop once per frame. Moving around the trajectory is an slow operation, so you should avoid it when you can. Here I would suggest something lighter such as:

```python
for tss in range(len(u.trajectory)):
    ...
```

Note that `tss` is not a good variable name and you should probably use something more explicit like `frame_index`.